content/blog/tyranny-of-nits/index.md

```diff
@@ -0,0 +1,208 @@
+Ultimately it comes down to the question of "what should be blocking".
+
+Let's make the **Leafwing Taxonomy of Code Concerns** (your PMs are more likely to use it if it has a fancy title).
+Years of painstaking fieldwork and sample collection have led me to the identification of four categories.
+
+1. **Clear mistakes:** the code doesn't work as intended.
+2. **Future work:** there's something cool that's related that we could or should do!
+3. **Polish problems:** documentation, formatting, micro-scale code quality problems.
+4. **Compromise conflict:** the change sacrifices some desirable property in favor of another one, but the reviewer doesn't agree that it's the right choice.
+
+Which of these should be blocking?
+
+Clear mistakes should be blocking: your code should work, and should have the tests to prove it.
+If the reviewer can produce a failing test case, and that test either wouldn't have failed before or wouldn't be more than an hour to fix, the change should be blocked until it's fixed.
+When it *can't* be, it evolves into a compromise conflict.
+
+Future work shouldn't block.
+Just make a follow-up issue and go talk about it over there to avoid derailing the thread.
+If you *can* make your PRs smaller: do.
+If you absolutely can't ship a "half-finished" feature to users (you probably can), then put it behind a feature flag and merge the PR anyways rather than exploding its scope.
+Merging and moving on maintains momentum and morale.
+
+What about polish problems?
+Well, yes and no!
+These are definitionally uncontroversial improvements (or sidegrades): why not have more of a good thing?
+Better docs will make users happy, and cleaner code will make life better for contributors.
+But these aren't *free*: the more time and energy you take in review, the more momentum you sap.
+
+The compromise I suggest is to let the automation guide you.
+If a polish problem can be automatically detected, make it blocking.
+Then, the author can fix and validate the problem without burning energy on feedback cycles.
+But if it can't be automated, focus on maintaining a pretty lenient bar for quality.
+Use Github's suggestion feature in your code reviews, or make PRs to their branch, and make it painless for the author to clean things up.
+When it comes time to merge: apply any suggestions or clean it up yourself.
+Don't let polish nits kill useful features and bug fixes!
+
+Compromise conflicts are where it gets tricky: there *is* no right answer, so it's very easy for discussion to go around in a circle rehashing disagreement until somone burns out and quits the thread or project.
+Last woman standing! Very Darwinian, but incredibly wasteful and "how far is each advocate from throwing up their hands" is a poor grounds on which to make technical decisions.
```

Comment:
Would be nice to flesh this out a bit on how to resolve this properly. Benevolent dictators, voting, etc.